apptools/io/h5/tests/test_file.py

```diff
@@ -509,15 +509,15 @@ def test_attribute_iteration_methods():
         attrs['count'] = 42
         attrs['alpha'] = 0xff
 
-        items = attrs.items()
+        items = list(attrs.items())
 
         assert all(isinstance(x, tuple) for x in items)
 
         # unfold the pairs
-        keys, vals = map(list, zip(*items))
+        keys, vals = [list(item) for item in zip(*items)]
```

Comment:
No idea why this was originally written in such a complicated way. Just write `keys, vals = zip(*items)`. It's a common Python idiom and everything.

Reply:
FWIW, I'm really not a fan of the `zip(*stuff)` idiom: it's got a nasty not-immediately-obvious corner case where `stuff` is empty. Fine in this case, but then it gets re-used in places where it's not fine and people don't notice the corner case until it's too late.

```
>>> d = {'a': 1, 'b': 2}
>>> keys, values = zip(*d.items())  # fine
>>> d = {}
>>> keys, values = zip(*d.items())  # not fine
```

Reply:
This doesn't detract from @jwiggins' comment, of course.

One possible reason for the `list` is that otherwise `keys` and `values` are tuples. I haven't looked at the context, so I don't know if that matters in this particular case, but that's the usual reason for the extra `map(list, ...)` construct.

Reply:
`keys`, `vals` end up as tuples when we just use `= zip(*items)`, which is not expected as we then try comparing to lists. I'm keeping the `[list(item) ... ]` code instead of updating other code to test against tuples or wrap `keys`/`vals` with `list` later in the code.

Reply:
Now that you've reminded me that `zip` returns tuples, that explains why it was initially written as `map(list, zip(*items))`.